packages/framework/src/Component/Doctrine/GroupedScalarHydrator.php

```diff
@@ -51,6 +51,8 @@ protected function gatherGroupedScalarRowData(&$data)
             $fieldName = $cacheKeyInfo['fieldName'];
 
             if (isset($cacheKeyInfo['isScalar'])) {
+                $type  = $cacheKeyInfo['type'];
+                $value = $type->convertToPHPValue($value, $this->_platform);
```

Comment:
What kind of sorcery is this???

Couple of lines later, there is ` $value = $type ? $type->convertToPHPValue($value, $this->_platform) : $value;` Why don't we check what the `$type` is?

Reply:
This code has been copy pasted from \Doctrine\ORM\Internal\Hydration\AbstractHydrator::gatherRowData:288-289

Reply:
Wow, I still don't understand it.... So if it works... why not :-)